### PR TITLE
Per pr previews

### PR DESCRIPTION
**Pull request by xjensen**
Per cagov/odi-engineering#53, implement per-PR preview sites. 

Adds a new GitHub Action to push PR'd branches to appropriate location in S3.

### CODE REVIEW

**Comment by xjensen** [comment]
Bringing this out of draft. It's now ready for review.

File: .github/workflows/eleventy_build_pr.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        with:
+          node-version: 14.17.0
+      - name: Build 11ty
+        run: |
+          mkdir dist
+          npm ci --production
+          npm run build
+      - name: Write robots.txt
+        run: |
+          echo 'User-agent: *' > docs/robots.txt
```
Comment (aaronhans) [inline .github/workflows/eleventy_build_pr.yml:28]: nice

File: .github/workflows/eleventy_build_pr.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        uses: jakejarvis/s3-sync-action@v0.5.1
+        with:
+          args: --acl public-read --follow-symlinks --delete
+        env:
+          AWS_S3_BUCKET: 'pr.drought.ca.gov'
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          AWS_REGION: 'us-west-1'   # optional: defaults to us-east-1
+          SOURCE_DIR: ./docs # only move built directory
+          DEST_DIR: /pr/$URLSAFE_BRANCH_NAME
```
Comment (aaronhans) [inline .github/workflows/eleventy_build_pr.yml:54]: cool

**Comment by aaronhans** [comment]
Pretty cool stuff Jon!

I see how it is writing to specific directories on S3 but I don't get how it becomes browseable at $URLSAFE_BRANCH_NAME.pr.drought.ca.gov would you please explain how that lets you browse the new directory as /

**Comment by xjensen** [comment]
@aaronhans The magic comes from a setting in the CloudFront distribution for this bucket. Under **Behaviors**, I've added a Lambda@Edge function for "Viewer Request".  This function rewrites the path from which CloudFront pulls the file from S3.

It's a simple function, under 10 lines of JS. You can check out the code in AWS Lambda for us-east-1 in our account. (Unfortunately, Lambda@Edge functions must go in the us-east-1 region.)

There's a simple test in place. Check out the contents of the bucket (pr.drought.ca.gov), and then visit the following URLs.

https://pr.drought.ca.gov/
https://alpha.pr.drought.ca.gov/
https://beta.pr.drought.ca.gov/

I'll need to provide more documentation, for sure. Lemme know if any questions or concerns.

**Comment by xjensen** [comment]
Also, I'm wondering if we should translate this into a CloudFormation template. This could automate set up of S3, CloudFront, etc. for new sites.

**Review by aaronhans — APPROVED** [review]
Nice! Thanks for explaining the browsable domain part

**Comment by aaronhans** [comment]
This is excellent, thanks for getting this figured out with so little code!

Yes templatizing this infrastructure creation is a good idea.

**Pull request merged**